usr/share/rear/layout/prepare/GNU/Linux/100_include_partition_code.sh

```diff
@@ -26,25 +26,15 @@ create_disk() {
     local component disk size label junk
     read component disk size label junk < <(grep "^disk $1 " "$LAYOUT_FILE")
 
-    ### Disks should be block devices.
-    [[ -b "$disk" ]]
-    BugIfError "Disk $disk is not a block device."
-
-    ### Find out the actual disk size.
-    local disk_size=$( get_disk_size $(get_sysfs_name "$disk") )
-
-    [[ "$disk_size" ]]
-    BugIfError "Could not determine size of disk $disk, please file a bug."
-
-    [[ $disk_size -gt 0 ]]
-    StopIfError "Disk $disk has size $disk_size, unable to continue."
-
     cat >> "$LAYOUT_CODE" <<EOF
 
 #
 # Code handling disk '$disk'
 #
 
+### Disks should be block devices.
+[ -b "$disk" ] || BugError "Disk $disk is not a block device."
+
```

Comment:
@pcahyna 
I had never thought before if it is OK or wrong to call Error()
(also via BugError() indirectly) within he diskrestore.sh script
so I expected that this already happens somewhere directly
or indirectly in our code.
I did not check what actually happens in case of Error()
inside diskrestore.sh.

Reply:
By the way:

In general it is bad to error out the whole program from inside a function.
Normally a function should return a failure value to its caller
and let the caller decide what to do in this case.

Of course there are exceptions for example the function
assert_ISO_FILE_SIZE_LIMIT in lib/output-functions.sh currently at
https://github.com/rear/rear/blob/master/usr/share/rear/lib/output-functions.sh#L28